src/main/java/org/dependencytrack/resources/v1/TeamResource.java

```diff
@@ -318,7 +320,7 @@ public Response regenerateApiKey(
     }
 
     @POST
-    @Path("/key/{key}/comment")
+    @Path("/key/{publicId}/comment")
```

Comment:
This would be a breaking change in the API. Can we deprecate endpoints accepting a full API key instead, and create separate endpoints that accept the public ID?

I think you should be able to call the same logic in both endpoint variants, just that you need to extract the public ID from the API key first for the deprecated endpoints.

Reply:
> [...] cause how, should the backend know which to use?

Yeah, we'd either:

* Have to be creative when it comes to naming the endpoints since yes, `/key/{value}/comment` is already taken.
* Leverage regex. [`@Path`](https://jakarta.ee/specifications/restful-ws/3.0/apidocs/jakarta/ws/rs/path#value()) accepts expressions. New API keys should always have three `_`, for example.

> I simply would keep the one endpoint, and determine there, if its a key or a public ID and add in the param description, that you also can use Keys for Legacy reasons

That of course also works.